ext/opcache/jit/zend_jit_arm64.dasc

```diff
@@ -172,6 +172,11 @@ static bool arm64_may_use_adrp(const void *addr)
 	return 0;
 }
 
+static bool arm64_may_encode_imm12(const int64_t val)
+{
+	return (val >= 0 && (val < (1<<12) || !(val & 0xffffffffff000fff)));
```

Comment:
Should use either CMP_IMM or drop the constant?

Reply:
Thanks for your comment. But I'm afraid I didn't quite get your point.

Do you mean, we'd better change  `val < (1<<12)` to either `val <= CMP_IMM`, or `val < 0x1000`, since `1<<12` is kind of abundant computation?

Reply:
I have updated the patch. Would you mind taking another look? Thanks! @nikic

Reply:
Could  you help to take another round of review on the latest commit when you have spare time? Thanks. @nikic